tests/unit/utils/test_version.py

```diff
@@ -950,3 +953,74 @@ def test_opengl_vendor():
     """Simply call version.opengl_vendor() and see if it doesn't crash."""
     pytest.importorskip("PyQt5.QtOpenGL")
     return version.opengl_vendor()
+
+
+class HTTPPostStub(QObject):
+
+    """A stub class for HTTPClient.
+
+    Attributes:
+        url: the last url send by post()
+        data: the last data send by post()
+    """
+
+    success = pyqtSignal(str)
+    error = pyqtSignal(str)
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.url = None
+        self.data = None
+
+    def post(self, url, data=None):
+        self.url = url
+        self.data = data
+
+
+@pytest.fixture
+def pbclient():
+    http_stub = HTTPPostStub()
+    client = pastebin.PastebinClient(http_stub)
+    return client
+
+
+def test_pastebin_version(pbclient, monkeypatch):
+    """Test version.pastebin_version() twice."""
+    patches = {
+        '_path_info': lambda: {'PATH DESC': 'PATH NAME'},
+        '_uptime': lambda: datetime.timedelta(hours=1, minutes=23, seconds=45),
+    }
+
+    for name, val in patches.items():
+        monkeypatch.setattr('qutebrowser.utils.version.' + name, val)
+
+    version.pastebin_version(pbclient)
+    pbclient.success.emit("test")
+    assert version.pastebin_url == "test"
+
+    version.pastebin_version(pbclient)
+    assert version.pastebin_url == "test"
+
+
+def test_pastebin_version_error(pbclient, caplog, monkeypatch):
+    """Test version.pastebin_version() with errors."""
+    patches = {
+        '_path_info': lambda: {'PATH DESC': 'PATH NAME'},
+        '_uptime': lambda: datetime.timedelta(hours=1, minutes=23, seconds=45),
+    }
+
+    for name, val in patches.items():
+        monkeypatch.setattr('qutebrowser.utils.version.' + name, val)
+
+    version.pastebin_url = None
+    with caplog.at_level(logging.ERROR):
```

Comment:
Using `message_mock` should be easier here - IIRC you said you wanted to look at that, did you run into trouble with that approach?

Reply:
It's just that caplog already contained all the information required to test the failure - the message output is in caplog.records[0]